code/datums/components/mutant_hands.dm

```diff
@@ -0,0 +1,173 @@
+ * but giving them powerful weapons to infect people
+ *
+ * It is suggested that the item path supplied has NODROP (and likely DROPDEL),
+ * but nothing's preventing you from not having that.
+ *
+ * If they lose or gain hands, new mutant hands will be created immediately.
+ *
+ * Does not override nodrop items that already exist in hand slots.
+ * However if those nodrop items are lost, will immediately create a new mutant hand.
+ */
+/datum/component/mutant_hands
+	// First come, first serve
+	dupe_mode = COMPONENT_DUPE_UNIQUE
+	/// The item typepath that we insert into the parent's hands
+	var/obj/item/mutant_hand_path = /obj/item/mutant_hand
+
+/datum/component/mutant_hands/Initialize(obj/item/mutant_hand_path = /obj/item/mutant_hand)
+	if(!ishuman(parent))
+		return COMPONENT_INCOMPATIBLE
+
+	src.mutant_hand_path = mutant_hand_path
+
+/datum/component/mutant_hands/RegisterWithParent()
+	// Give them a hand before registering ANYTHING just so it's clean
+	INVOKE_ASYNC(src, PROC_REF(apply_mutant_hands))
+
+	RegisterSignals(parent, list(COMSIG_CARBON_POST_ATTACH_LIMB, COMSIG_CARBON_POST_REMOVE_LIMB), PROC_REF(try_reapply_hands))
+	RegisterSignal(parent, COMSIG_MOB_EQUIPPED_ITEM, PROC_REF(mob_equipped_item))
+	RegisterSignal(parent, COMSIG_MOB_UNEQUIPPED_ITEM, PROC_REF(mob_dropped_item))
+
+/datum/component/mutant_hands/UnregisterFromParent()
+	UnregisterSignal(parent, list(
+		COMSIG_CARBON_POST_ATTACH_LIMB,
+		COMSIG_CARBON_POST_REMOVE_LIMB,
+		COMSIG_MOB_EQUIPPED_ITEM,
+		COMSIG_MOB_UNEQUIPPED_ITEM,
+	))
+
+	// Remove all their hands after unregistering everything so they don't return
+	INVOKE_ASYNC(src, PROC_REF(remove_mutant_hands))
+
+/**
+ * Tries to give the parent mob mutant hands.
+ *
+ * * If a hand slot is empty, places the mutanthand type into their hand.
+ * * If a hand slot is filled with a nodrop item, it will do nothing.
+ * * If a hand slot is filled with a non-nodrop item, drops the item to the ground.
+ * * If a hand slot is filled with a hand already, does nothing.
+ */
+/datum/component/mutant_hands/proc/apply_mutant_hands()
+	var/mob/living/carbon/human/human_parent = parent
+	for(var/obj/item/hand_slot as anything in human_parent.held_items)
+		// This slot is already a mutant hand
+		if(istype(hand_slot, mutant_hand_path))
+			continue
+		message_admins("Checking [hand_slot || "nothing"]")
+		// This slot is not empty
+		// Yes the held item lists contains nulls to represent empty hands
+		// It saves us a /item cast by using as anything in the loop
+		if(!isnull(hand_slot))
+			if(HAS_TRAIT(hand_slot, TRAIT_NODROP) || (hand_slot.item_flags & ABSTRACT))
+				message_admins("[hand_slot] is nodrop or abstract")
+				// There's a nodrop / abstract item in the way of putting a mutant hand in
+				// It can stay, for now, but if it gets dropped / unequipped we'll swoop in to replace the slot
+				continue
+			message_admins("[hand_slot] can be dropped safely")
```

Comment:
Are these `message_admins` stuff for debugging, or do you actually want to pass this feedback along to admins? seems quite vague to be useful in production imo, would need to include more context to be useful.